### PR TITLE
Extra static assert

### PR DESCRIPTION
**Pull request by adetaylor**

### CODE REVIEW

**Comment by adetaylor** [comment]
@bsilver8192 Would you take a look here? Is this what you were envisaging for the remaining problems in #835?

File: integration-tests/tests/integration_test.rs [hunk 1]
```diff
@@ -5287,6 +5287,64 @@ fn test_error_generated_for_pod_with_nontrivial_move_constructor() {
...
+                if (*flag) {
+                    fprintf(stderr, \"DestructorFlag is already set\\n\");
+                    abort();
+                }
+                *flag = true;
+                // Note we deliberately do NOT clear the value of `flag`, to catch Rust calling
+                // this destructor twice.
+            }
+
+            bool *flag = nullptr;
```
Comment (bsilver8192) [inline integration-tests/tests/integration_test.rs:5312]: You could remove all of this (and just call it `NotTriviallyDestructible`), and everything in `rs` below. Just `~DestructorFlag() {}` makes it nontrivial, everything else is extra lines that aren't supposed to compile.

**Review by bsilver8192 — APPROVED** [review]
LGTM, that addresses the problem. Noticed a few redundant parts of the tests you could remove if you want.

File: integration-tests/tests/integration_test.rs (hunk 1 above)
Comment (adetaylor) [reply integration-tests/tests/integration_test.rs:5312]: Done, thanks

File: integration-tests/tests/integration_test.rs [hunk 2]
```diff
@@ -5287,6 +5287,64 @@ fn test_error_generated_for_pod_with_nontrivial_move_constructor() {
     run_test_expect_fail("", hdr, rs, &["take_a"], &["A"]);
 }
```
[the remark this answers is not in the document]
Comment (adetaylor) [reply integration-tests/tests/integration_test.rs:5289]: I think I'll keep them. They're more concise and they test the case where the POD type _is_ used in the cxx::bridge, so it doesn't hurt to have both.

**Review by bsilver8192 — APPROVED** [review]

**Pull request merged**